addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/ssl/UniFiTrustManagerProvider.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.unifi.internal.ssl;
+
+import javax.net.ssl.X509ExtendedTrustManager;
+
+import org.eclipse.smarthome.io.net.http.TlsTrustManagerProvider;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ *
+ * The {@link UniFiTrustManagerProvider} is an implementation of {@link TlsTrustManagerProvider} which provides an
+ * instance of {@link UniFiTrustManagerProvider} for all servers that use a certificate with the common name equal to
+ * "UniFi" (<code>CN=UniFi</code>).
+ *
+ * @author Matthew Bowman - Initial contribution
+ */
+@Component
+public class UniFiTrustManagerProvider implements TlsTrustManagerProvider {
+
+    @Override
+    public String getHostName() {
+        return "UniFi";
+    }
+
+    @Override
+    public X509ExtendedTrustManager getTrustManager() {
+        return new UniFiTrustManager();
```

Comment:
I guess you should not return a new instance on every call.

Reply:
Implemented this as a singleton